lib/egis/table.rb

```diff
@@ -17,11 +17,13 @@ class Table
     DEFAULT_OPTIONS = {format: :tsv}.freeze
 
     def initialize(database, name, schema, location, options: {},
+                   client: Egis::Client.new,
                    partitions_generator: Egis::PartitionsGenerator.new,
                    table_ddl_generator: Egis::TableDDLGenerator.new,
-                   output_downloader: Egis::OutputDownloader.new,
+                   output_downloader: Egis::OutputDownloader.new(client.aws_s3_client),
```

Comment:
If we went event further down this road, passing `configuration`, we could avoid that direct dependency on `client` but I won't be so anal about it. ;) 
```suggestion
                   output_downloader: Egis::OutputDownloader.new(configuration),
```